File: apps/api/Models/PageSchema.cs
using System.Text.Json.Serialization;

namespace Api.Models;

public class PageSchema
{
    public string Id { get; set; } = string.Empty;
    public string Slug { get; set; } = string.Empty;
    public string TemplateId { get; set; } = string.Empty;
    public string TenantId { get; set; } = string.Empty;
    public SeoMeta Seo { get; set; } = new();
    public ThemeConfig Theme { get; set; } = new();
    public List<Section> Sections { get; set; } = new();
}

public class SeoMeta
{
    public string Title { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string? OgImage { get; set; }
    public string? Canonical { get; set; }
    public bool? NoIndex { get; set; }
}

public class ThemeConfig
{
    public string PrimaryColor { get; set; } = "#6366f1";
    public string SecondaryColor { get; set; } = "#8b5cf6";
    public string? AccentColor { get; set; }
    public string? BackgroundColor { get; set; }
    public string? TextColor { get; set; }
    public string? PrimaryFont { get; set; }
    public string? SecondaryFont { get; set; }
}

public class Section
{
    public string Id { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool Visible { get; set; } = true;
    public string Label { get; set; } = string.Empty;
    public List<ComponentNode> Components { get; set; } = new();
}

public class ComponentNode
{
    public string Id { get; set; } = string.Empty;
    public string Type { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool Visible { get; set; } = true;
    public string? Label { get; set; }
    public string Path { get; set; } = string.Empty;
    public bool Editable { get; set; }
    public bool? Repeatable { get; set; }
    public ComponentNode? RepeatableTemplate { get; set; }
    public int? MinItems { get; set; }
    public int? MaxItems { get; set; }
    public object? Value { get; set; }
    public TypographyConfig? Typography { get; set; }
    public ColorConfig? Color { get; set; }
    public LinkConfig? Link { get; set; }
    public List<AttributeNode>? Attributes { get; set; }
    public List<ComponentNode>? Children { get; set; }
    public ValidationRules? Validation { get; set; }
}

public class TypographyConfig
{
    public string? FontSize { get; set; }
    public string? FontWeight { get; set; }
    public string? TextAlign { get; set; }
    public string? FontFamily { get; set; }
    public string? LetterSpacing { get; set; }
    public string? LineHeight { get; set; }
    public string? TextTransform { get; set; }
}

public class ColorConfig
{
    public string? Text { get; set; }
    public string? Background { get; set; }
    public string? Border { get; set; }
}

public class LinkConfig
{
    public string Href { get; set; } = string.Empty;
    public string? Target { get; set; }
    public string? Rel { get; set; }
    public string? AriaLabel { get; set; }
}

public class AttributeNode
{
    public string Id { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string Type { get; set; } = "string";
    public object Value { get; set; } = string.Empty;
    public bool Editable { get; set; }
    public string? Label { get; set; }
}

public class ValidationRules
{
    public bool? Required { get; set; }
    public int? MinLength { get; set; }
    public int? MaxLength { get; set; }
    public string? Pattern { get; set; }
    public string? ErrorMessage { get; set; }
    public List<string>? AcceptedFileTypes { get; set; }
    public int? MaxFileSize { get; set; }
}

public class ImageValue
{
    public string Src { get; set; } = string.Empty;
    public string Alt { get; set; } = string.Empty;
    public int? Width { get; set; }
    public int? Height { get; set; }
}

File: apps/landing/src/types/page-schema.ts
export interface PageSchema {
  id: string;
  slug: string;
  templateId: string;
  tenantId: string;
  seo: SeoMeta;
  theme: ThemeConfig;
  sections: Section[];
}

export interface SeoMeta {
  title: string;
  description: string;
  ogImage?: string;
  canonical?: string;
  noIndex?: boolean;
}

export interface ThemeConfig {
  primaryColor: string;
  secondaryColor: string;
  accentColor?: string;
  backgroundColor?: string;
  textColor?: string;
  primaryFont?: string;
  secondaryFont?: string;
}

export interface Section {
  id: string;
  type: SectionType;
  order: number;
  visible: boolean;
  label: string;
  components: ComponentNode[];
}

export type SectionType =
  | 'header' | 'hero' | 'features' | 'testimonials'
  | 'pricing' | 'faq' | 'cta' | 'gallery'
  | 'team' | 'stats' | 'contact' | 'footer' | 'custom';

export interface ComponentNode {
  id: string;
  type: ComponentType;
  order: number;
  visible: boolean;
  label?: string;
  path: string;
  editable: boolean;
  repeatable?: boolean;
  repeatableTemplate?: ComponentNode;
  minItems?: number;
  maxItems?: number;
  value?: ComponentValue;
  typography?: TypographyConfig;
  color?: ColorConfig;
  link?: LinkConfig;
  attributes?: AttributeNode[];
  children?: ComponentNode[];
  validation?: ValidationRules;
}

export type ComponentType =
  | 'text' | 'heading' | 'paragraph' | 'image' | 'video'
  | 'icon' | 'button' | 'badge' | 'richtext'
  | 'group' | 'list' | 'grid'
  | 'card' | 'feature-card' | 'testimonial-card'
  | 'pricing-card' | 'stat-card' | 'team-member'
  | 'faq-item' | 'nav-item' | 'social-link';

export type ComponentValue = string | number | boolean | ImageValue | VideoValue;

export interface ImageValue {
  src: string;
  alt: string;
  width?: number;
  height?: number;
}

export interface VideoValue {
  src: string;
  poster?: string;
  autoplay?: boolean;
  muted?: boolean;
}

export interface TypographyConfig {
  fontSize?: string;
  fontWeight?: string;
  textAlign?: 'left' | 'center' | 'right' | 'justify';
  fontFamily?: 'primary' | 'secondary';
  letterSpacing?: string;
  lineHeight?: string;
  textTransform?: 'uppercase' | 'lowercase' | 'capitalize' | 'normal';
}

export interface ColorConfig {
  text?: string;
  background?: string;
  border?: string;
}

export interface LinkConfig {
  href: string;
  target?: '_blank' | '_self';
  rel?: string;
  ariaLabel?: string;
}

export interface AttributeNode {
  id: string;
  name: string;
  type: 'string' | 'number' | 'boolean';
  value: string | number | boolean;
  editable: boolean;
  label?: string;
}

export interface ValidationRules {
  required?: boolean;
  minLength?: number;
  maxLength?: number;
  pattern?: string;
  errorMessage?: string;
  acceptedFileTypes?: string[];
  maxFileSize?: number;
}
